### PR TITLE
Literate OneDShallowWater example

### PR DESCRIPTION
**Pull request by navidcy**

### CODE REVIEW

**Comment by navidcy** [comment]
a preview of the literated version of the OneDShallowWater example is here: https://fourierflows.github.io/FourierFlowsDocumentation/previews/PR177

@apaloczy could you review this PR?

**Comment by glwagner** [comment]
Wow. This is so nice.

File: examples/OneDShallowWaterGeostrophicAdjustment.jl [hunk 1]
```diff
@@ -1,184 +1,428 @@
...
+#md # This example can be run online via [![](https://mybinder.org/badge_logo.svg)](@__BINDER_ROOT_URL__/generated/OneDShallowWaterGeostrophicAdjustment.ipynb). 
+#md # Also, it can be viewed as a Jupyter notebook via [![](https://img.shields.io/badge/show-nbviewer-579ACA.svg)](@__NBVIEWER_ROOT_URL__/generated/OneDShallowWaterGeostrophicAdjustment.ipynb).
+# 
+#
+#
 # This example solves the linear 1D rotating shallow water equations
-# for the depth-integrated velocities u and v and for the
-# surface elevation η in a fluid with constant depth h,
-# gravitational acceleration g and Coriolis parameter f:
+# for the $u(x, t)$, $v(x, t)$ and the surface surface elevation $\eta(x, t)$,
```
Comment (apaloczy) [inline examples/OneDShallowWaterGeostrophicAdjustment.jl:9]: for the depth-averaged velocities $u(x, t)$, $v(x, t)$ and the surface elevation $\eta(x, t)$

File: examples/OneDShallowWaterGeostrophicAdjustment.jl [hunk 2]
```diff
@@ -1,184 +1,428 @@
...
-# jets around the Gaussian bump, while the smaller-scale noise
-# propagates around as inertia-gravity waves.
-
-using FourierFlows, PyPlot
-using FFTW: rfft, irfft, fft, ifft
+# Rotation introduces the deformation length scale, $L_d = \sqrt{g H}/f$. 
+# Disturbances with length scales much smaller than $L_d$ don't "feel" 
+# the rotation and propagate as inertia-gravity waves. Disturbances with 
+# length scales comparable or larger than $L_d$ should be approximately 
+# in geostrophic balance, i.e., the Coriolis accelaration
```
Comment (apaloczy) [inline examples/OneDShallowWaterGeostrophicAdjustment.jl:30]: acceleration

File: examples/OneDShallowWaterGeostrophicAdjustment.jl [hunk 3]
```diff
@@ -1,184 +1,428 @@
+# # Linear rotating shallow water dynamics
+#
+#md # This example can be run online via [![](https://mybinder.org/badge_logo.svg)](@__BINDER_ROOT_URL__/generated/OneDShallowWaterGeostrophicAdjustment.ipynb).
```
Comment (apaloczy) [inline examples/OneDShallowWaterGeostrophicAdjustment.jl:3]: Binder and Jupyter notebook links seem to be broken

File: examples/OneDShallowWaterGeostrophicAdjustment.jl (hunk 1 above)
Comment (navidcy) [reply examples/OneDShallowWaterGeostrophicAdjustment.jl:9]: Andre, should have mentioned that I rewrote the equations for the normal u, v, (not depth integrated). Equations now read:
∂u/∂t - fv = -g ∂η/∂x - D
∂v/∂t + fu = - D
∂η/∂t + H ∂u/∂x = - D

Are these OK?

File: examples/OneDShallowWaterGeostrophicAdjustment.jl (hunk 3 above)
Comment (navidcy) [reply examples/OneDShallowWaterGeostrophicAdjustment.jl:3]: Yeap thanks for catching this; I moved the documentation in a different repository and I have to figure out what the new urls are...

File: examples/OneDShallowWaterGeostrophicAdjustment.jl [hunk 4]
```diff
@@ -1,184 +1,428 @@
...
+vars = Vars(dev, grid)
+equation = Equation(dev, params, grid)
+
+prob = FourierFlows.Problem(equation, stepper, dt, grid, vars, params, dev)
+nothing #hide
+
+# ## Setting initial conditions
+
+# For initial condition we take the fluid at rest ($u=v=0$). The free surface elevation
+# is perturbed from its rest position ($\eta=0$); the disturbance we imposeis a Gaussian
```
Comment (apaloczy) [inline examples/OneDShallowWaterGeostrophicAdjustment.jl:242]: imposes

File: examples/OneDShallowWaterGeostrophicAdjustment.jl (hunk 1 above)
Comment (apaloczy) [reply examples/OneDShallowWaterGeostrophicAdjustment.jl:9]: Yes, I agree that depth-averaged is better. Looks good.

File: examples/OneDShallowWaterGeostrophicAdjustment.jl (hunk 3 above)
Comment (navidcy) [reply examples/OneDShallowWaterGeostrophicAdjustment.jl:3]: ad04e13 fixes this

**Pull request merged**